src/System.DirectoryServices/tests/System/DirectoryServices/DirectoryServicesTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Collections;
+using Xunit;
+
+namespace System.DirectoryServices.Tests
+{
+    public class DirectoryServicesTests
+    {
+        internal static bool IsLdapConfigurationExist => LdapConfiguration.Configuration != null;
+        
+        [ConditionalFact(nameof(IsLdapConfigurationExist))]
+        public void TestOU() // adding and removing organization unit
+        {
+			using (DirectoryEntry de = new DirectoryEntry(LdapConfiguration.Configuration.LdapPath,
+														    LdapConfiguration.Configuration.UserName,
+														    LdapConfiguration.Configuration.Password,
+														    LdapConfiguration.Configuration.AuthenticationTypes))
+            {
+                string ouName = "NetCoreDevs";
+                CreateOU(de, ouName, ".Net Core Developers Unit");
+                try
+                {
+                    SearchOUByName(de, ouName);
+                }
+                finally 
+                {
+                    // Clean up the added ou
+                    DeleteOU(de, ouName);
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(IsLdapConfigurationExist))]
+        public void TestCN() // adding and removing organization unit
+        {
+			using (DirectoryEntry de = new DirectoryEntry(LdapConfiguration.Configuration.LdapPath,
+														    LdapConfiguration.Configuration.UserName,
+														    LdapConfiguration.Configuration.Password,
+														    LdapConfiguration.Configuration.AuthenticationTypes))
+            {
+                DirectoryEntry rootOU = CreateOU(de, "CoreFxRootOU", "CoreFx Root OU");
+                try
+                {
+                    DirectoryEntry child1OU = CreateOU(rootOU, "CoreFxChild1OU", "CoreFx Child OU 1");
+                    DirectoryEntry child2OU = CreateOU(rootOU, "CoreFxChild2OU", "CoreFx Child OU 2");
+
+                    CreateCN(child1OU, "user.ou1.1", "User 1 is in CoreFx ou 1", "1 111 111 11111");
+                    CreateCN(child1OU, "user.ou1.2", "User 2 is in CoreFx ou 1", "1 222 222 22222");
+
```

Comment:
I suppose we'll add some test logging in the future so it's clear which of these sub steps failed when the test fails?

Reply:
XUnit will catch any thrown exception and have it in the logs.